contracts/gateway/AxelarAmplifierGateway.sol

```diff
@@ -63,93 +78,117 @@ contract AxelarAmplifierGateway is IAxelarAmplifierGateway {
         string calldata sourceAddress,
         bytes32 payloadHash
     ) external override returns (bool valid) {
-        bytes32 key = _getIsContractCallApprovedKey(commandId, sourceChain, sourceAddress, msg.sender, payloadHash);
-        valid = _axelarAmplifierGatewayStorage().approvals[key];
-
-        if (valid) {
-            delete _axelarAmplifierGatewayStorage().approvals[key];
+        valid = _validateContractCall(commandId, sourceChain, sourceAddress, payloadHash);
+    }
 
-            emit ContractCallExecuted(commandId);
-        }
+    function validateMessage(
+        string calldata messageId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes32 payloadHash
+    ) external override returns (bool valid) {
+        bytes32 commandId = keccak256(bytes(messageId));
+        valid = _validateContractCall(commandId, sourceChain, sourceAddress, payloadHash);
     }
 
     /***********\
     |* Getters *|
     \***********/
 
     function isCommandExecuted(bytes32 commandId) public view override returns (bool) {
-        return _axelarAmplifierGatewayStorage().commands[commandId];
+        return _storage().commands[commandId];
     }
 
     /**********************\
     |* External Functions *|
     \**********************/
 
-    function execute(bytes calldata batch) external {
-        (bytes memory data, bytes memory proof) = abi.decode(batch, (bytes, bytes));
+    function execute(SignedCommandBatch calldata signedBatch) external {
+        CommandBatch calldata batch = signedBatch.batch;
+        bytes memory data = abi.encode(batch);
```

Comment:
Hey maybe we don't need to encode it again but can slice it out from `msg.data`?